src/test/java/edu/harvard/iq/dataverse/api/AdminIT.java

```diff
@@ -463,6 +463,46 @@ public void testConvertOAuthUserToBuiltin() throws Exception {
         assertEquals(200, deleteSuperuser.getStatusCode());
 
     }
+    
+    @Test
+    public void testChangeAuthenticatedUserIdentifier() {
+        Response createSuperuser = UtilIT.createRandomUser();
+        String superuserUsername = UtilIT.getUsernameFromResponse(createSuperuser);
+        String superuserApiToken = UtilIT.getApiTokenFromResponse(createSuperuser);
+        Response toggleSuperuser = UtilIT.makeSuperUser(superuserUsername);
+        toggleSuperuser.then().assertThat()
+                .statusCode(OK.getStatusCode());
+        
+        Response createUser = UtilIT.createRandomUser();
+        createUser.prettyPrint();
+        assertEquals(200, createUser.getStatusCode());
+        String usernameOfUser = UtilIT.getUsernameFromResponse(createUser);
+        
+        String newUsername = "newUser_" + UtilIT.getRandomString(4);
+        Response changeAuthIdResponse = UtilIT.changeAuthenticatedUserIdentifier(usernameOfUser, newUsername, superuserApiToken);
+        changeAuthIdResponse.prettyPrint();
+        changeAuthIdResponse.then().assertThat()
+                .statusCode(OK.getStatusCode());
+        
+        String newUsernameBad = ""; //one character, should fail before bean validation even
+        Response changeAuthIdResponseBad = UtilIT.changeAuthenticatedUserIdentifier(usernameOfUser, newUsernameBad, superuserApiToken);
+        changeAuthIdResponseBad.prettyPrint();
+        changeAuthIdResponseBad.then().assertThat()
+                .statusCode(BAD_REQUEST.getStatusCode());
+        
+        String newUsernameBad2 = "z"; //one character, should fail bean validation
```

Comment:
should we test trying to change it into a user that already exists?

Reply:
Yeah good call

Reply:
Actually, beyond just testing, maybe make sure we catch it as an user friendly error.